42.FragmentShaderPathTracer/common.glsl

```diff
@@ -686,10 +688,14 @@ bool closestHitProgram(in uint depth, in uint _sample, inout Ray_t ray, inout nb
 
 void main()
 {
+    const ivec2 imageExtents = nbl_glsl_swapchain_transform_preTransformExtents(cameraData.swapchainTransform, imageSize(outImage));
     const ivec2 coords = getCoordinates();
-    const vec2 texCoord = getTexCoords();
+    // In an example that renders vertices to screen you apply transform it to the projection matrix (given triangle vertices you map them to screen). Remember rasterization is a scatter op.
+    // However in an example such as this, where you're casting rays, you're finding out worldspace positions for screenspace pixels so transform needs to be applied in reverse. Raycasting/post process is a gather op.
+    vec2 texCoord = (vec2(nbl_glsl_swapchain_transform_preTransform(cameraData.swapchainTransform, coords, imageSize(outImage))) / vec2(imageExtents));
+    texCoord.y = 1.0 - texCoord.y;
 
-    if (false == (all(lessThanEqual(ivec2(0),coords)) && all(greaterThan(imageSize(outImage),coords)))) {
+    if (false == (all(lessThanEqual(ivec2(0),coords)) && all(greaterThan(imageExtents,coords)))) {
```

Comment:
Should we use `texCoord` here instead of `coords`?
Assuming we have a
```
transform = NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_90
[width,height] = [10, 20]
coords range are from [0, 0] to [10, 20] (right?)
```
then 
`nbl_glsl_swapchain_transform_preTransformExtents` will give us `[20, 10]` and
then we will only render to [10, 10] 

Of course it doesn't change anything right now when we only test HORIZONTAL_180_MIRROR